pkg/reconciler/reconciler.go

```diff
@@ -33,14 +36,66 @@ func (r Updater) Reconcile(ctx context.Context, c client.Client, scheme *runtime
 				r.resource.GetObjectKind().GroupVersionKind().String(), err)
 		}
 	}
-
 	owner := fmt.Sprintf("%s/%s", r.resourceOwner.GetNamespace(), r.resourceOwner.GetName())
-	if err := c.Patch(ctx, r.resource, client.Apply, client.ForceOwnership, client.FieldOwner(owner)); err != nil {
-		return fmt.Errorf("%s/%s (%s): updater failed to patch: %w",
-			r.resource.GetNamespace(), r.resource.GetName(),
-			r.resource.GetObjectKind().GroupVersionKind().String(), err)
+
+	// HACK: workarounds around the issue that the operator sometimes fails
+	// to create a rolebinding  because a role that was created earlier could not be found.
+	// E.g. while creating a RoleBinding "valid-loglevel-alertmanager"
+	// Patch fails with Error "roles.rbac.authorization.k8s.io 'valid-loglevel-alertmanager' not found"
+	//
+	// The workaround is to get the resource that was applied by Patch and
+	// on failure to get the resource back, reapply the Patch
+
+	var patchError error
+	pollErr := wait.PollImmediate(5*time.Second, 30*time.Second, func() (bool, error) {
+
+		// these const makes is easier to understand the return statements
+		const done = true
+		const retry = false
+
+		// reset patchError for every attempt
+		patchError = nil
+
+		ns := r.resource.GetNamespace()
+		name := r.resource.GetName()
+		gvk := r.resource.GetObjectKind().GroupVersionKind()
+
+		// failing to patch shouldn't be retried since it is a genuine error
+		if err := c.Patch(ctx, r.resource, client.Apply, client.ForceOwnership, client.FieldOwner(owner)); err != nil {
+			patchError = fmt.Errorf("%s/%s (%s): updater failed to patch: %w", ns, name, gvk.String(), err)
+			return done, nil
+		}
+
+		// Get requires a client.Object but we can't use r.resource since it may
+		// need to be reapplied, so deepcopy the resource and set the GVK
+		dup, ok := r.resource.DeepCopyObject().(client.Object)
```

Comment:
@simonpasquier @jan--f, Do you know of a better way that type asserting the copy? 
I have seem similar approach in controller-runtime code as well but if you know of a better way, I am all ears